File: 34-find-first-and-last-position-of-element-in-sorted-array/34-find-first-and-last-position-of-element-in-sorted-array.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
function sendPos(nums, target, di, n){
    var l = 0;
    var r = n-1;
    var returnable = -1;
    while(l <= r){
        var mid = l+(r-l)//2;
        if(nums[mid] === target){
            if(di){
                returnable = mid;
                r = mid-1;
            }
            else{
                returnable = mid;
                l = mid+1;
            }
        }
        else if(nums[mid] >  target){
            r = mid-1;
        }
        else{
            l = mid+1;
        }
    }
    return returnable;
}
var searchRange = function(nums, target) {
    var start = sendPos(nums, target, true, nums.length);
    var end = sendPos(nums, target, false, nums.length);
    var ans = [start, end];
    return ans;
};

File: 34-find-first-and-last-position-of-element-in-sorted-array/34-find-first-and-last-position-of-element-in-sorted-array.cpp
class Solution {
public:
    vector<int> searchRange(vector<int>& nums, int target) {
        int strt = 0;
        int end = nums.size();
        vector<int>ans;
        while(strt < end){
            int mid = (strt+end)/2;
            if(nums[mid] >= target){
                end = mid;
            }
            else{
                strt = mid+1;
            }
        }
        int l = strt;
        strt = 0, end = nums.size();
        while(strt < end){
            int mid = (strt+end)/2;
            if(nums[mid] <= target){
                strt = mid+1;
            }
            else{
                end = mid;
            }
        }
        int r = strt;
        if(l == r){
            ans.push_back(-1);
            ans.push_back(-1);
        }
        else{
            ans.push_back(l);
            ans.push_back(r-1);
        }
        return ans;
    }
};